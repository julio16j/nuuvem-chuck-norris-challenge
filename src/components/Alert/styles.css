.alert {
    padding: 1rem 1rem;
    margin-top: 1rem;
    margin-bottom: 1rem;
    border: 1px solid transparent;
    border-radius: 0.25rem;
    cursor: pointer;
  }
  .error {
    color: white !important;
    background-color: var(--negative) !important;
    border-color: var(--gray) !important;
  }
  .closebtn {
    margin-left: 15px;
    color: white;
    font-weight: bold;
    float: right;
    font-size: 22px;
    line-height: 20px;
    cursor: pointer;
    transition: 0.3s;
  }
  .alert:hover {
    background-color: var(--shaded-gray) !important;
    box-shadow: 0 1px 6px 0 #171717;
    border-color: rgba(223,225,229,0);
  }

  .hide {
    display: none;
  }