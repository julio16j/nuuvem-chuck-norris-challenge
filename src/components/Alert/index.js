import React from "react";
import './styles.css'
import typeAlertEnumArray from "../../utils/typeAlertEnum"
function Alert({ type, message, isShow=false, handleClose }) {
  if (!typeAlertEnumArray.includes(type)) {
    type = ""
  }
  return (
    <div className={!isShow ? 'hide' : `${type} alert`} onClick={handleClose} >
      {message}
      <span className='closebtn'>
        &times;
      </span>
    </div>
  );
}
export default Alert
