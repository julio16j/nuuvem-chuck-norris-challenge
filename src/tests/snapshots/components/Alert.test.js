import renderer from "react-test-renderer";
import Alert from "../../../components/Alert";
import React from "react";

it("Alert Component renders correctly", () => {
  const component = renderer
    .create(<Alert message={"Test Message"} />)
    .toJSON();
  expect(component).toMatchSnapshot();
});

it("Alert Component renders correctly", () => {
    const component = renderer
      .create(<Alert message={"Test Message"} isShow={true} />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

it("Alert Component renders correctly", () => {
    const component = renderer
      .create(<Alert message={"Test Message"} isShow={true} type="error" />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });