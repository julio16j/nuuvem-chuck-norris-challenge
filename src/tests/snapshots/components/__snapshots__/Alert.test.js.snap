// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Component renders correctly 1`] = `
<div
  className="hide"
>
  Test Message
  <span
    className="closebtn"
  >
    ×
  </span>
</div>
`;

exports[`Alert Component renders correctly 2`] = `
<div
  className=" alert"
>
  Test Message
  <span
    className="closebtn"
  >
    ×
  </span>
</div>
`;

exports[`Alert Component renders correctly 3`] = `
<div
  className="error alert"
>
  Test Message
  <span
    className="closebtn"
  >
    ×
  </span>
</div>
`;
